docs/content/docs/reference/react/hooks.mdx

```diff
@@ -0,0 +1,393 @@
+        </div>
+      )}
+    </div>
+  )
+}
+```
+
+### Email + password authentication
+
+```js
+const { signIn, isLoading, needsVerification, isSuccess, isError, error } =
+  useEmailPasswordSignIn(email?: string, password?: string)
+```
+
+- `signIn: (email?: string, password?: string) => void` will try to authenticate. The email/password arguments will take precedence over the possible state values used when creating the hook.
+- `isLoading: boolean` returns `true` when the action is executing, `false` when the action result is available.
+- `needsVerification: boolean` returns `true` if the user email is still pending verification.
+- `isSuccess: boolean` returns `true` if the user has successfully authenticated. Returns `false` in case or error or if the new email needs to be verified first.
+- `isError: boolean` returns `true` if an error occurred.
+- `error?: {status: number, error: string, message: string}` provides details about the error.
+
+#### Usage
+
+```jsx
+import { useState } from 'react'
+import { useEmailPasswordSignIn } from '@nhost/react'
+
+const Component = () => {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const { signIn, isLoading, isSuccess, needsVerification, isError, error } =
+    useEmailPasswordSignIn(email, password)
+  return (
+    <div>
+      <input value={email} onChange={(event) => setEmail(event.target.value)} placeholder="Email" />
+      <input
+        value={password}
+        onChange={(event) => setPassword(event.target.value)}
+        placeholder="Password"
+      />
+      <button onClick={signUp}>Register</button>
+      {isSuccess && <div>Authentication suceeded</div>}
+      {needsVerification && (
+        <div>
+          Your email is still pending verification. Check your mailbox and follow the instructions
+          to activate your account.
+        </div>
+      )}
+    </div>
+  )
+}
+```
+
+### Oauth providers
+
+```js
+import { useProviderLink } from '@nhost/react'
+
+const Component = () => {
+  const { github } = useProviderLink()
+  return <a href={github}>Authenticate with GitHub</a>
+}
+```
+
+### Passwordless email authentication
+
+```js
+const { signIn, isLoading, isSuccess, isError, error } =
+  useEmailPasswordlessSignIn(email?: string, options?: PasswordlessOptions)
+```
+
+- `signIn: (email?: string) => void` will send a magic link to the given email The email argument will take precedence over the the possible state value used when creating the hook.
+- `isLoading: boolean` returns `true` when the action is executing, `false` when the action result is available.
+- `isSuccess: boolean` returns `true` if the magic link email user has successfully send.
+- `isError: boolean` returns `true` if an error occurred.
+- `error?: {status: number, error: string, message: string}` provides details about the error.
+- `PasswordlessOptions`:
```

Comment:
Are these options specific to "passwordless" (given the name `PasswordlessOptions`). I think they are generic to other sign-up methods too?

Reply:
yes, they are the same as for email+password, I renamed to `Options` so it simplifies the documentation